include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -372,35 +413,35 @@ struct erfInv_helper<FloatingPoint NBL_PARTIAL_REQ_BOT(concepts::FloatingPointSc
 {
 	static FloatingPoint __call(NBL_CONST_REF_ARG(FloatingPoint) _x)
 	{
-		FloatingPoint x = clamp<FloatingPoint>(_x, -0.99999, 0.99999);
+		FloatingPoint x = clamp<FloatingPoint>(_x, FloatingPoint(-0.99999), FloatingPoint(0.99999));
 
-		FloatingPoint w = -log_helper<FloatingPoint>::__call((1.0 - x) * (1.0 + x));
+		FloatingPoint w = -log_helper<FloatingPoint>::__call((FloatingPoint(1.0) - x) * (FloatingPoint(1.0) + x));
 		FloatingPoint p;
 		if (w < 5.0)
 		{
-			w -= 2.5;
-			p = 2.81022636e-08;
-			p = 3.43273939e-07 + p * w;
-			p = -3.5233877e-06 + p * w;
-			p = -4.39150654e-06 + p * w;
-			p = 0.00021858087 + p * w;
-			p = -0.00125372503 + p * w;
-			p = -0.00417768164 + p * w;
-			p = 0.246640727 + p * w;
-			p = 1.50140941 + p * w;
+			w -= FloatingPoint(2.5);
+			p = FloatingPoint(2.81022636e-08);
+			p = FloatingPoint(3.43273939e-07) + p * w;
+			p = FloatingPoint(-3.5233877e-06) + p * w;
+			p = FloatingPoint(-4.39150654e-06) + p * w;
+			p = FloatingPoint(0.00021858087) + p * w;
+			p = FloatingPoint(-0.00125372503) + p * w;
+			p = FloatingPoint(-0.00417768164) + p * w;
+			p = FloatingPoint(0.246640727) + p * w;
+			p = FloatingPoint(1.50140941) + p * w;
 		}
 		else
 		{
-			w = sqrt_helper<FloatingPoint>::__call(w) - 3.0;
-			p = -0.000200214257;
-			p = 0.000100950558 + p * w;
-			p = 0.00134934322 + p * w;
-			p = -0.00367342844 + p * w;
-			p = 0.00573950773 + p * w;
-			p = -0.0076224613 + p * w;
-			p = 0.00943887047 + p * w;
-			p = 1.00167406 + p * w;
-			p = 2.83297682 + p * w;
+			w = sqrt_helper<FloatingPoint>::__call(w) - FloatingPoint(3.0);
+			p = FloatingPoint(-0.000200214257);
+			p = FloatingPoint(0.000100950558) + p * w;
+			p = FloatingPoint(0.00134934322) + p * w;
+			p = FloatingPoint(-0.00367342844) + p * w;
+			p = FloatingPoint(0.00573950773) + p * w;
+			p = FloatingPoint(-0.0076224613) + p * w;
+			p = FloatingPoint(0.00943887047) + p * w;
+			p = FloatingPoint(1.00167406) + p * w;
+			p = FloatingPoint(2.83297682) + p * w;
```

Comment:
no just a C-cast will not work

You need the `NBL_FLOAT64_LITERAL` macro because it adds a platform dependant suffix

> So a double bag of `float_t(NBL_FLOAT64_LITERAL(0.3232983928))` is needed

The reason is that the constant literal itself is not fp32, so its already at reduced/truncated precision by the time you C-cast it to float64 (promoting it)

You still have the old problem https://github.com/Devsh-Graphics-Programming/Nabla/pull/821#discussion_r1947978451

p.S. maaaybe because you're assigning straight to `const FloatingPoint` the `NBL_FLOAT64_LITERAL` macro is enough.